server/auth_function/test/lambda_function_test.py

```diff
@@ -70,6 +72,70 @@ def test_create_user_if_not_found(
         sql.Literal(test_cognito_user_id),
         sql.Literal(test_idp_username),
     )
+
+    cursor.execute(replaced_query)
+    count = cursor.fetchone()[0]
+    assert count == 1
+
+
+@pytest.mark.parametrize(
+    "cognito_event",
+    [
+        "login_event_bceid.json",
+    ],
+    indirect=True,
+)
+def test_create_bceid_user_if_not_found_test_with_business_guid(
+    db_pg_transaction, cognito_event, cognito_context, test_user_properties
+):
+    cursor = db_pg_transaction.cursor()
+
+    # shorter variables
+    test_idp_type_code = test_user_properties["idp_type_code"]
+    test_idp_user_id = test_user_properties["idp_user_id"]
+    test_cognito_user_id = test_user_properties["cognito_user_id"]
+    test_idp_username = test_user_properties["idp_username"]
+    test_idp_business_guid = test_user_properties["idp_business_id"]
+
+    # make sure the user doesn't exist
+    user_query = sql.SQL(
+        """SELECT user_name from app_fam.fam_user where
+            user_name = {0}"""
+    ).format(sql.Literal(test_idp_username))
+
+    cursor.execute(user_query)
+    results = cursor.fetchall()
+    LOGGER.debug(f"results: {results}")
+    assert len(results) == 0
+    assert results == []
+
+    # simulate a login
+    lambda_function.lambda_handler(cognito_event, cognito_context)
+
+    # verify that the user exists now
+    cursor.execute(user_query)
+    results = cursor.fetchall()
+    LOGGER.debug(f"results: {results}")
+    assert len(results) == 1
+    assert results[0][0] == test_idp_username
+
+    # validate that there is one user in the database with the properties from
+    # the incoming event, the user is created with the business_guid
+    raw_query = """select count(*) from app_fam.fam_user where
+        user_type_code = {} and
+        user_guid = {} and
+        cognito_user_id = {} and
+        user_name = {} and
+        business_guid = {};"""
+
+    replaced_query = sql.SQL(raw_query).format(
+        sql.Literal(test_idp_type_code),
+        sql.Literal(test_idp_user_id),
+        sql.Literal(test_cognito_user_id),
+        sql.Literal(test_idp_username),
+        sql.Literal(test_idp_business_guid),
```

Comment:
Test that for business bceid user login with attribute `custom:idp_business_id` configured readable and writable, it will be added to the database